AppController/djinn.rb

```diff
@@ -5944,71 +5945,55 @@ def get_request_info(app_id, secret)
   end
 
 
-  # Searches through the requests to scale up and down each application in this
-  # AppScale deployment, and determines if machines need to be terminated due
-  # to excess capacity.
+  # Examines the given node, for which recently an AppServer had been downscaled,
+  # and checks if it still has any other AppServers running. If not, then the
+  # machine is terminated due to excess capacity.
   #
   # Args:
-  #   all_scaling_votes: A Hash that maps each appid (a String) to the votes
-  #     received to scale the app up or down (an Array of Strings).
+  #   app_name: A String that indicates which application was being scaled down.
+  #   node_ip: A String that indicates the private IP of the node being scaled down.
   # Returns:
-  #   An Integer that indicates how many nodes were added to this AppScale
-  #   deployment. A negative number indicates that that many nodes were
-  #   removed from this AppScale deployment.
-  def examine_scale_down_requests(all_scaling_votes)
-    # First, only scale down in cloud environments.
+  #   An Integer that indicates whether or not this current node was downscaled.
+  #   A negative number indicates that the node was removed from this AppScale
+  #   deployment.
+  def examine_scaling_down_instances(app_name, node_ip)
+    # Only scale down in cloud environments.
     unless is_cloud?
       Djinn.log_debug("Not scaling down VMs, because we aren't in a cloud.")
       return 0
     end
 
+    # Only scale down, if we have more machines than the user specified 
+    # minimum number of machines.
     if @nodes.length <= Integer(@options['min_images']) or @nodes.length <= 1
```

Comment:
If the user specifies ips_layout instead of min/max, what gets defined for min_images?